cpp/models/ide_secir/model.h

```diff
@@ -0,0 +1,174 @@
+     * @brief Computes total number of Deaths for the current last time in m_populations.
+     * 
+     * Number is stored in m_populations.
+     *
+     */
+    void compute_deaths();
+
+    /**
+     * @brief Computes total number of Recovered for the current last time in m_populations.
+     * 
+     * Number is stored in m_populations.
+     *
+     */
+    void compute_recovered();
+
+    /**
+     * @brief Get the size of the compartment specified in idx_InfectionState at the current last time in m_populations.
+     * 
+     * Calculation is reasonable for all compartments except S, R, D. 
+     * Therefore, we have alternative functions for those compartments.
+     *
+     * @param[in] idx_InfectionState Specifies the considered InfectionState
+     * @param[in] idx_IncomingFlow Specifies the index of the infoming flow to InfectionState in m_transitions. 
+     * @param[in] idx_TransitionDistribution1 Specifies the index of the first relevant TransitionDistribution, 
+     *              related to a flow from the considered InfectionState to any other InfectionState.
+     *              This index is also used for related probability.
+     * @param[in] idx_TransitionDistribution2 Specifies the index of the second relevant TransitionDistribution, 
+     *              related to a flow from the considered InfectionState to any other InfectionState (in most cases to Recovered). 
+     *              Necessary related probability is calculated via 1-probability[idx_TransitionDistribution1].
+     * @param[in] dt Time discretization step size.
+     */
+    void compute_compartment(Eigen::Index idx_InfectionState, Eigen::Index idx_IncomingFlow,
+                             int idx_TransitionDistribution1, int idx_TransitionDistribution2, ScalarType dt);
+
+    /**
+     * @brief Sets all values of remaining compartments (compartments apart from S, R, D) for the current last timestep in m_populations.
+     *
+     * New values are stored in m_populations. Most values are computed via the function get_size_of_compartments().
+     * 
+     * @param[in] dt Time discretization step size.
+     */
+    void other_compartments_current_timestep(ScalarType dt);
+
+    ParameterSet parameters{}; ///< ParameterSet of Model Parameters.
+    /* Attention: m_populations and m_transitions do not necessarily have the same number of time points due to the initialization part. */
+    TimeSeries<ScalarType>
+        m_transitions; ///< TimeSeries containing points of time and the corresponding number of transitions.
+    TimeSeries<ScalarType>
+        m_populations; ///< TimeSeries containing points of time and the corresponding number of people in defined InfectionState%s.
+
+private:
+    ScalarType m_forceofinfection{0}; ///< Force of infection term needed for numerical scheme.
+    ScalarType m_N{0}; ///< Total population size of the considered region.
+    ScalarType m_deaths_before{0}; ///< Deaths before start of simulation (at time -m_dt).
```

Comment:
remove as member?

Reply:
If we remove this as member we have to give Dead_before as argument to update_forceofinfection and thus also to the Simulate Class. At least I don't see another way right now. In my opinion this doesn't make the code more readable as we only need Dead_before in update_forceofinfection() once in the initalization. But maybe there is a better solution to this.